Library/Homebrew/formulary.rb

```diff
@@ -311,16 +323,28 @@ def ruby_source_checksum
       end
     end
 
-    T.cast(klass, T.class_of(Formula)).loaded_from_api = true
-    mod.const_set(class_s, klass)
+    klass = T.cast(klass, T.class_of(Formula))
+
+    mod.const_set(class_name, klass)
 
     cache[:api] ||= {}
     cache[:api][name] = klass
   end
 
-  def self.resolve(name, spec: nil, force_bottle: false, flags: [])
+  sig { params(name: String, spec: Symbol, force_bottle: T::Boolean, flags: T::Array[String]).returns(Formula) }
```

Comment:
```suggestion
  sig { params(name: String, spec: T.nilable(Symbol), force_bottle: T::Boolean, flags: T::Array[String]).returns(Formula) }
```

no, given these are set to `nil` below?

Reply:
No, the type should be `Symbol`. You should only ever call it with e.g. `spec: :stable`, but never explicitly with `spec: nil`. The `T.unsafe(nil)` is only there to distinguish if the parameter is absent or not.